<a href="https://colab.research.google.com/github/trevorwjames/DS-Unit-1-Sprint-2-Statistics/blob/master/%20/module3%20/%20LS_DS_124_Introduction_to_Bayesian_Inference_Assignment_Trevor_James.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




## Assignment - Code it up!

We used pure math to apply Bayes Theorem to drug tests. Now write Python code to reproduce the results! This is purposefully open ended - you'll have to think about how you should represent probabilities and events. You can and should look things up.

Specific goals/targets:

### 1) Write a function 

`def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):` 

You should only truly need these three values in order to apply Bayes Theorem. In this example, imagine that individuals are taking a breathalyzer test with an 8% false positive rate, a 100% true positive rate, and that our prior belief about drunk driving in the population is 1/1000. 
 - What is the probability that a person is drunk after one positive breathalyzer test?
 - What is the probability that a person is drunk after two positive breathalyzer tests?
 - How many positive breathalyzer tests are needed in order to have a probability that's greater than 95% that a person is drunk beyond the legal limit?


In [0]:
# function for the probability that the person taking the breathalizer test is actually drunk

def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):
  prob_non_drunk = 1-prob_drunk_prior
  
  numerator = true_positive_rate * prob_drunk_prior
  denominator = true_positive_rate * prob_drunk_prior + false_positive_rate * prob_non_drunk
  return numerator / denominator 

In [9]:
prob_drunk_given_positive(.001, .08, 1)

0.012357884330202669

The probability that a person is drunk after their first breathalizer test is 
1.2%

In [13]:
prob_drunk_given_positive(.0123, .08, 1)

0.13469709579920275

The probability that the person is drunk after their first breathalizer test is 13.5%, we need to keep testing



In [14]:
prob_drunk_given_positive(.134, .08, 1)

0.6591892955529319

In [16]:
prob_drunk_given_positive(.659, .08, 1)

0.960249460861456

after our fourth test we can now be 96% certain that the breathalizer test will give us a true result. 


In [12]:
for ci in prob_drunk_given_positive(ci, .08, 1):
  prob_drunk_given_positive(.001, .08, 1) = ci

SyntaxError: ignored

In [21]:
for ci in 

TypeError: ignored

### 2) In your own words, summarize the difference between Bayesian and Frequentist statistics

If you're unsure where to start, check out [this blog post of Bayes theorem with Python](https://dataconomy.com/2015/02/introduction-to-bayes-theorem-with-python/).

In basic terms Bayesian Statistics uses current bias in order to calculate the proabability of something being true. We need a stat of the way the world currently works in order to figure probability of an outcome. Here the key is to decrease the degrees of freedom as well as have an accurate starting statistic. 

Frequentist Statistics uses Null Hypothesis in order to understand how the world works. We are looking for sample sizes that can compare back to the population. Here an increased sample size is the key to getting better stats. 

### 3) Use the following Template to help come up with ideas for your Build Sprint Project: 



---

## Idea 1:

### You
What do you care about?

What do you know about?

What decisions do you face?




I care about the environment, I took some classes in high school as well as originally looking to major in environmental science. We face a major decision as a world to look into way to better treat the environment or learn to implement tech to better coincide with the planet.

### Seven templates
Training Kit (https://learn.lambdaschool.com/ds/module/recedjanlbpqxic2r) explains the seven templates from Priceonomics.

Can you apply the templates to your topics?

**Geographic Variation:**

**Trend related to the news:**

**Who does that?:**

**Answering a question people care about:**

**Valuable to businesses:**

**What's the most popular?:**

**Cost/Money rankings:**

Geographic variation: Different areas have difference concerns when it comes to the environment they interact with. Arizona worries about hotter summers, polution regarding trying to stay cool, and clean water. Where I grew up in the northeast there are concerns with winters vs sumemrs affecting the natural habitat. I have a love for winter sports that would be affected with lighter winters

Trend Related to news: Huge politcal topic currently. How are candidates leveraging it in their policital discussion during campaigning. How could it affect the weather and how it is reported. Are weather people biased at different times of year. 

Who does that?: Where in the country people are interested in climate change, what countries are more green thinking than others. are there states where people believe climate change is a "belief" rather than science. 

Answering a question people will care about: How will this affect me? everyday people care about the environment although most are selfish with what they CAN do or how it will really affect them

Valuable to a Business: Environmental concerns can be hugely impactful to a busienss when it comes to how the business is seen around the world. Also creating less waste can save money and time. 

Whats the most popular?: What is the most popular way to "go green". What is the most popular way they believe they are contributing to the environment. 

Cost/Money Rankings: Breaking down the cost or the savings involved with working against climate change. What is it cost affective to buy a electric car? What does it look like to have solar on your house financially?



### Misconceptions

What misconceptions do people have about your topic?

The fact that our current president does believe in climate change. That one persons efforts will not make a difference. That we will either create tech to save us, or that we wont be able to do anything about it. 

### Examples

What data storytelling example inspires you?

Could you do a new hypothesis, for the same question?

Could you do a new question, for the same topic?

Could you do a new topic, with the same "style"?

Liv Johnsons story here inspires me. If I could find a way to see the financial cost of solar, as well as the current cost of electricity for americans it could be cool to compare the availability to solar energy to how much it were to cost the average american


###Data

Where could you search for data about your topic?

Solar Panel Data on Priceonomics:https://priceonomics.com/the-most-solar-places-in-america/

National Survey on Energy and Environment: https://datasetsearch.research.google.com/search?query=environment&docid=O%2FegCJJNWFhQIJm0AAAAAA%3D%3D

Survey of Public attitudes and behaviours toward the environment: https://datasetsearch.research.google.com/search?query=environment&docid=MPSnArJVOOTGAqeuAAAAAA%3D%3D

Climate Change: Earth surface temperature Data: https://datasetsearch.research.google.com/search?query=Climate%20Change&docid=r32LBb1oqrQepe0eAAAAAA%3D%3D

---

## Idea 2:

### You
What do you care about?

What do you know about?

What decisions do you face?



I care about health and fitness, I majored in exercise and movement science becasue I have a belief that people should have the ability to control their health through simple principles. In our world we face a decision to take personal responsibility for the health of ourselves rather than relying on a healthcare system that is put in place to be extremely reactive rather than teaching its citizens how to stay healthy. 

### Seven templates
Training Kit (https://learn.lambdaschool.com/ds/module/recedjanlbpqxic2r) explains the seven templates from Priceonomics.

Can you apply the templates to your topics?

**Geographic Variation:**

**Trend related to the news:**

**Who does that?:**

**Answering a question people care about:**

**Valuable to businesses:**

**What's the most popular?:**

**Cost/Money rankings:**

Geographic Variation: How do the ideas of health and fitness differ in regions around the world? What are certain areas doing to combat this global problem vs other areas?

Trend Related to news: Whom are the most suceptible population to contracting the Coronavirus and experiencing hardship or possible death? How do people with metabolic syndrome put pressure on our healthcare system? How does that affect the poliical climate?

Who Does that?: How has the world of health anf fitness changed over the past 100 years? the past 10 years? How has the advent of social media challenged what people belive when it comes to fitness or health?

Answering questions people care about: The basic idea of what is true and what is not? How are people suppose to access help in regard to Health and fitness. 

Valuable to Business: Healthcare businesses - how can they change their policies to better serve the people. How can businesses leverage a healthier workforce in order to save money on healthcare, more productive, or less sick days? So many things here

What's Most popular: Right now the comparison of health and fitness to deaths from coronavirus, or even reportin symptoms of COVID.

Cost/Money rankings: A huge that comes to mind when thinking about health and fitness and finances are the market for products and ability to forsee what is popular, where should you invest in health and fitness, etc. 


### Misconceptions

What misconceptions do people have about your topic?

- Eating healthy is expensive
- Going to a gym is expensive
- Cardio exercise keeps you the most healthy 
- It is not as important as people think

### Examples

What data storytelling example inspires you?

Could you do a new hypothesis, for the same question?

Could you do a new question, for the same topic?

Could you do a new topic, with the same "style"?

###Data

Where could you search for data about your topic?

Seeing some good results on this topic on kaggle, google seems tough to find actual data sets. 

Health examination survey : https://www.kaggle.com/cdc/national-health-and-nutrition-examination-survey?select=questionnaire.csv


## Resources

- [Worked example of Bayes rule calculation](https://en.wikipedia.org/wiki/Bayes'_theorem#Examples) (helpful as it fully breaks out the denominator)
- [Source code for mvsdist in scipy](https://github.com/scipy/scipy/blob/90534919e139d2a81c24bf08341734ff41a3db12/scipy/stats/morestats.py#L139)

## Stretch Goals:

- Go back and study the content from Modules 1 & 2 & 3 to make sure that you're really comfortable with them.
- Apply a Bayesian technique to a problem you previously worked (in an assignment or project work) on from a frequentist (standard) perspective
- Check out [PyMC3](https://docs.pymc.io/) (note this goes beyond hypothesis tests into modeling) - read the guides and work through some examples
- Take PyMC3 further - see if you can build something with it!